gui/notify.lua

```diff
@@ -64,6 +65,11 @@ end
 function NotifyOverlay:overlay_onupdate()
     local choices = {}
     local is_adv = dfhack.world.isAdventureMode()
+    if is_adv then
+        self.overlay_onupdate_max_freq_seconds=0
```

Comment:
I dunno how to refactor this because something as important as your suffocation bar needs to be delivered to the player as swiftly as possible. It's especially relevant for swimming.

Perhaps it should only be updated every tick? However there's no existing logic allowing it to be updated every tick currently.

Also, what specialized subclasses do you mean? They can't affect the frequency of updating of the overlay, can they?

Reply:
> Also, what specialized subclasses do you mean?

I meant `DwarfNotifyOverlay` and `AdvNotifyOverlay` below, but I think this will take some more work in the framework

For now, could you revert changes to this file and instead set `critical=true` on the two notification types in `notifications.lua`?

Reply:
how's this? https://github.com/DFHack/scripts/pull/1222/files